### PR TITLE
New StateResponsive widget

### PR DESCRIPTION
**Pull request by ejprok**
Makes setting up ViewState dependent UI easy

### CODE REVIEW

File: lib/ui/widgets/state_responsive.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
```
Comment (VictorUvarov) [inline lib/ui/widgets/state_responsive.dart]: This can be replaced with import 'package:flutter/widgets.dart';

**Review by VictorUvarov — CHANGES_REQUESTED** [review]
Looks good. Very promising widget to reduce view state checks. If you could replace the viewstate checks in the template with this widget I would be happy to merge.

File: lib/ui/widgets/state_responsive.dart (hunk 1 above)
Comment (ejprok) [reply lib/ui/widgets/state_responsive.dart]: Nice catch

**Comment by ejprok** [comment]
HomeView and PostDetailView now use the StateResponsive widget.  None of the other views look like they need it.

File: lib/ui/views/home_view.dart [hunk 2]
```diff
@@ -22,19 +23,33 @@ class HomeView extends StatelessWidget {
             transitionBetweenRoutes: false,
           ),
         ),
-        body: _Posts(),
+        body: _Body(),
```
Comment (VictorUvarov) [inline lib/ui/views/home_view.dart]: This build method is too short now, replace _Body with the stateResponsive widget and use the withConsumer constructor for the view model provider instead

**Review by VictorUvarov — CHANGES_REQUESTED** [review]
Looks good some minor changes. Also there is an unused import in home view for view state line 4

**Review by VictorUvarov — CHANGES_REQUESTED** [review]
Looks good just some small changes. Check for unused imports.

**Review by VictorUvarov — APPROVED** [review]
Nice job. Thanks

**Pull request merged**